scripts/configure-venue-access.ts

```diff
@@ -94,16 +74,28 @@ initFirebaseAdminApp(projectId);
       console.log(
         `Setting venues/${venueId}/access/${method} to {emails: ${emails}}...`
       );
-      await accessDocRef.set({
-        emails: mergeStringArrays(emails, (access as VenueAccessEmails).emails),
-      });
+      await accessDocRef.set(
+        {
+          emails: emails,
+        },
+        { merge: true }
+      );
       break;
 
     case VenueAccessMode.Codes:
+      const codes = fs
+        .readFileSync(accessDetail, "utf-8")
+        .split(/\r?\n/)
+        .forEach((line) => {
+          access.codes.push(line.trim());
+        });
       console.log(`Setting venues/${venueId}/access/${method}...`);
-      await accessDocRef.set({
-        codes: (access as VenueAccessCodes).codes,
-      });
+      await accessDocRef.set(
+        {
+          codes: codes,
+        },
+        { merge: true }
+      );
```

Comment:
Same concerns as called about with the `email` code above, RE: whether this will actually merge the `codes` data, or if it will overwrite that key, and if the latter, about  potentially using `arrayUnion`:

- https://firebase.google.com/docs/firestore/manage-data/add-data#update_elements_in_an_array

Reply:
The merge option parameter seems to be working fine without the need for arrayUnion.

Reply:
Awesome!